lib/presentation/core/widgets/authentication/form/text_fields/date_field.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+import '../../../../../../utils/extensions.dart';
+import 'custom_text_form_field.dart';
+
+class DateField extends StatefulWidget {
+  const DateField({
+    super.key,
+  });
+
+  @override
+  State<DateField> createState() => _DateFieldState();
+}
+
+class _DateFieldState extends State<DateField> {
+  TextEditingController dateInput = TextEditingController();
+  @override
+  Widget build(BuildContext context) {
+    return CustomTextFormField(
+      controller: dateInput,
+      prefixIcon: const Icon(Icons.calendar_today),
+      hintText: context.l10n.dateOfBirth,
+      readOnly: true,
+      onTap: () async {
+        final now = DateTime.now();
+        final pickedDate = await showDatePicker(
+            context: context,
+            initialDate: DateTime(now.year - 18, now.month, now.day),
```

Comment:
I really dislike any math inside widgets, even the simplest ones. I'd definitely separate it to give it more meaning.

Reply:
E.g. 
```dart
DateTime initialDate() {
  final now = DateTime.now();

  // Laws differ from country to country, therefore this separation
  // may come in handy later, when we will want to support more countries.
  const adulteryAge = 18;

  return DateTime(now.year - adulteryAge, now.month, now.day);
}
```